packages/components/menu/src/menu.ts

```diff
@@ -128,16 +128,37 @@ export default defineComponent({
     // methods
     const initMenu = () => {
       const activeItem = activeIndex.value && items.value[activeIndex.value]
-      if (!activeItem || props.mode === 'horizontal' || props.collapse) return
+      if (
+        !activeItem ||
+        props.mode === 'horizontal' ||
+        props.collapse ||
+        (props.uniqueOpened && !activeItemFirstOpend.value)
+      )
+        return
 
       const indexPath = activeItem.indexPath
 
       // 展开该菜单项的路径上所有子菜单
       // expand all subMenus of the menu item
-      indexPath.forEach((index) => {
-        const subMenu = subMenus.value[index]
-        subMenu && openMenu(index, subMenu.indexPath)
-      })
+      const openAllActiveItem = () => {
+        if (activeItemFirstOpend.value) activeItemFirstOpend.value = false
```

Comment:
The judgment condition here also seems unnecessary.
```suggestion
        activeItemFirstOpend = false
```